sn_networking/src/lib.rs

```diff
@@ -396,33 +390,25 @@ impl Network {
             );
 
             let res = self
-                .put_record_once(
-                    record.clone(),
-                    verify_store.clone(),
-                    optional_permit,
-                    quorum,
-                )
+                .put_record_once(record.clone(), verify_store.clone())
                 .await;
             if !matches!(res, Err(Error::FailedToVerifyRecordWasStored(_))) {
                 return res;
             }
 
-            // the permit will have been consumed above.
-            optional_permit = None;
-
             retries += 1;
         }
         Err(Error::FailedToVerifyRecordWasStored(record.key.into()))
     }
 
-    async fn put_record_once(
-        &self,
-        record: Record,
-        verify_store: Option<Record>,
-        starting_permit: Option<OwnedSemaphorePermit>,
-        quorum: Quorum,
-    ) -> Result<()> {
-        let mut _permit = starting_permit;
+    async fn put_record_once(&self, record: Record, verify_store: Option<Record>) -> Result<()> {
+        // get permit if semaphore supplied
+        let _permit = if let Some(semaphore) = self.concurrency_limiter.clone() {
```

Comment:
just to confirm so here we no longer `consume` the fetched permit, but leave it to be dropped when the whole function call completed ?

Reply:
we hold it until the PUT is done, which is what was happening before with an optionally passed permit. 

(which should have just be grabbed above)